src/btouch.cs

```diff
@@ -422,13 +422,14 @@ static int Main2 (string [] args)
 			// Keep source files at the end of the command line - csc will create TWO assemblies if any sources preceed the -out parameter
 			var cargs = new StringBuilder ();
 
+			// A bit of a hack
+			bool isCSC = compiler.Contains ("/Library/Frameworks/Mono.framework/Versions/Current/bin/csc") || 
+				     compiler.Contains ("/Library/Frameworks/Mono.framework/Commands/csc");
 			if (CurrentPlatform == PlatformName.MacOSX) {
-				// HACK
-				bool isCSC = compiler.Contains ("/Library/Frameworks/Mono.framework/Versions/Current/bin/csc");
 				if (!isCSC && !string.IsNullOrEmpty (net_sdk) && net_sdk != "mobile")
 					cargs.Append ("-sdk:").Append (net_sdk).Append (' ');
 			} else {
-				if (!string.IsNullOrEmpty (net_sdk))
+				if (!isCSC && !string.IsNullOrEmpty (net_sdk))
```

Comment:
I _think_ is assume that `csc` was the only option outside macOS, i.e. on windows
However the `isCSC` check won't work on windows

Reply:
I think the hack should remain under the `MacOSX` condition

Reply:
or maybe it's time to remove the hack and *require* `csc`
because if we don't test with `mcs` it's only a matter of time (generally pretty short) before it stops working

Reply:
A reverse check would work on windows (`isMCS`)

```csharp
var isMCS = compiler.EndsWith ("/mcs", StringComparison.Ordinal);
...
if (isMCS && !string.IsNullOrEmpty (net_sdk))
    ...
```

Reply:
wrong path delimiter ;-) and likely extension (.exe or .bat ...)

Reply:
Did we use mcs on windows before? For some reason I thought we'd always used csc (in which case the mcs check would have been a platform check too, precisely because of those reasons), but thinking about it I see I must have been mistaken. So NVM 😄

Reply:
It turns out we don't execute the generator on windows, we execute it on the mac (like we do for mtouch): https://gist.github.com/GouriKumari/2b500792f55fe206dcb468b10df2a652#file-gistfile1-txt-L2408.